spec/features/irs_attempts_api/event_tracking_spec.rb

```diff
@@ -30,12 +30,15 @@
     sign_in_user(user)
 
     events = irs_attempts_api_tracked_events
+    expected_event_types = ['email-and-password-auth', 'mfa-phone-verification-otp-sent']
 
-    expect(events.count).to eq(1)
-    event = events.first
-    expect(event.event_metadata[:email]).to eq(user.email)
-    expect(event.event_metadata[:success]).to eq(true)
-    expect(event.session_id).to eq('test-session-id')
+    received_event_types = []
+    events.each do |event|
+      received_event_types.push(event.event_type)
+    end
```

Comment:
This is perfectly fine Ruby, but you can make this much more compact, as just:

```ruby
received_event_types = events.collect { |e| e.event_type }
```

You can actually condense that a step further:


```ruby
received_event_types = events.map(&:event_type)
```

The `&:event_type` shortcut is a little ugly, but it essentially invokes `event_type` on every item.

Reply:
Done. Thank you so much!